scripts/install_dependencies/install_openvino_dependencies.sh

```diff
@@ -16,260 +16,352 @@
 
 set -e
 
+#===================================================================================================
+# Option parsing
+
+interactive=yes
+optional=yes
+extra=
+self_test=
+os=${os:-auto}
+
+while :; do
+    case $1 in
+        -h|-\?|--help)
+            echo "Options:"
+            echo "  -y      non-interactive run (off)"
+            echo "  -r      install required packages only (off)"
+            echo "  -e      add extra repositories (CentOS 7) (off)"
+            exit
+            ;;
+        -y) interactive= ;;
+        -r) optional= ;;
+        -e) extra=yes ;;
+        --selftest) self_test=yes ;;
+        *) break ;;
+    esac
+    shift
+done
+
+#===================================================================================================
+#  Selftest
+
+if [ -n "$self_test" ] ; then
+    for image in centos:7 ubuntu:18.04 ubuntu:20.04 ; do
+        for opt in "-h" "-y -r" "-y" "-y -e -s" ; do
+            echo "||"
+            echo "|| Test $image / '$opt'"
+            echo "||"
+            docker run -it -v `pwd`:/scripts:ro $image  bash /scripts/install_openvino_dependencies.sh $opt
```

Comment:
Consider adding `--rm` option.

> pwd

We need script's directory instead of current process working directory.

> /scripts/install_openvino_dependencies.sh

Consider using `${0##*/}` (basename) instead of `install_openvino_dependencies.sh`:
- https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
- https://stackoverflow.com/questions/22401091/bash-variable-substitution-vs-dirname-and-basename

---

```
SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
for ...
    docker run -it --rm -v ${SCRIPT_DIR}:/scripts:ro $image bash /scripts/${0##*/} $opt
done
```